src/lightning/pytorch/strategies/launchers/multiprocessing.py

```diff
@@ -210,7 +209,6 @@ def _check_torchdistx_support(self) -> None:
 
     def get_extra_results(self, trainer: "pl.Trainer") -> Dict[str, Any]:
         """Gather extra state from the Trainer and return it as a dictionary for sending back to the main process.
-        To avoid issues with memory sharing, we cast the data to numpy.
```

Comment:
casting to numpy is done to prevent issues with memory sharing so I won't change this as a part of the refactor PR.

Reply:
Okay I will update this

Reply:
I have reverted the changes for this file